File: anplayer-plugin/src/main/java/com/xeinebiu/anplayer/plugin/media/extractor/extractor/AuthorExtractor.kt
package com.xeinebiu.anplayer.plugin.media.extractor.extractor

import com.xeinebiu.anplayer.plugin.media.extractor.exception.AuthorNotFoundException
import com.xeinebiu.anplayer.plugin.media.extractor.model.Author
import com.xeinebiu.anplayer.plugin.media.extractor.model.descriptor.MediaDescriptor

/**
 * Extractor for a single Author
 */
abstract class AuthorExtractor : Extractor<MediaDescriptor>() {

    /**
     * Returns the author
     */
    @Throws(AuthorNotFoundException::class)
    abstract fun getAuthor(): Author
}

File: src/lib/extractor/author.extractor.ts
import { Extractor } from './extractor';
import { Album, AuthorDetail, Media } from '../model';

/**
 * Extractor for the content of a single author (channel, artist,
 * studio, etc.).
 *
 * Returned by `AnPlayerExtractor.getAuthorExtractor(author)`. The root
 * `next()` method yields the author's published items, standalone tracks,
 * albums, or a mix of both.
 */
export type AuthorExtractor = Extractor<
    Media | Album,
    {
        /**
         * Returns the full detail record for this author (biography, artwork,
         * popularity, aliases, etc.).
         *
         * Throw `AuthorNotFoundError` if the author no longer exists.
         */
        getAuthor(): Promise<AuthorDetail>;
    }
>;
